### PR TITLE
Fixed PHPStan warnings for branch "next" (PHP 8.1)

### PR DESCRIPTION
**Pull request by fballiano**
The "next" branch now runs on PHP 8.1, this generated a few new PHPStan errors, this PR aims to fix them.

The only weird change that I've done is the removal of the deprecated method `getChildren()` in `app/code/core/Mage/Api/Model/Wsdl/Config/Element.php`, it was deprecated in https://github.com/OpenMage/magento-lts/pull/1613 and, since this is the "next" branch, I think we can remove it.

### CODE REVIEW

**Comment by sreichel** [comment]
We should think about to split phpstan-baseline ...

- a base file  ... for all branches
- include additional files for php7/8 specific errors

**Comment by fballiano** [comment]
I don't know if it's possible (since I don't know much about phpstan) but I don't know if it's worth the efforst since when next becomes main then the v20 will be frozen etc etc

File: app/code/core/Mage/Media/Model/File/Image.php
Comment (kiatng) [inline app/code/core/Mage/Media/Model/File/Image.php:1]: Shouldn't this be in main branch as well?

File: lib/Varien/Io/Ftp.php
Comment (kiatng) [inline lib/Varien/Io/Ftp.php:1]: I think this should be in main branch as well.

File: app/code/core/Mage/Media/Model/File/Image.php
Comment (fballiano) [reply app/code/core/Mage/Media/Model/File/Image.php:1]: @kiatng https://www.php.net/GdImage that class doesn't exist in php7 so I can't put it in the main branch :-(

Comment (kiatng) [reply app/code/core/Mage/Media/Model/File/Image.php:1]: But it's only docblock.

Comment (fballiano) [reply app/code/core/Mage/Media/Model/File/Image.php:1]: Yes but phpstan doesnt like it vecause the workflows run on php7 so phpstan reports it as a non existing class

**Review by kiatng — APPROVED** [review]

**Review by elidrissidev — APPROVED** [review]

**Comment by fballiano** [comment]
sorry guys, I had to push the composer.lock because it didn't pass the "validate" check (there was a conflict merging main to next a few days ago).

if all tests pass I'll merge right away without bothering you more, sorry!

**Pull request merged**